scheduler/local_scd_server.py

```diff
@@ -0,0 +1,320 @@
+
+            # Skip line is used for special time radar lines
+            if skip_line:
+                skip_line = False
+                continue
+
+            # We've hit the SCD notes and no longer need to parse
+            if "# Notes:" in line:
+                break
+
+            # Skip only white space lines
+            if not line.strip():
+                continue
+
+            #Lines starting with '#' are comments
+            if line[0] == "#":
+                continue
+
+            items = line.split()
+
+            #First line is month and year
+            if idx == 0:
+                continue
+
+            start_day = items[0][0:2]
+            start_hr = items[0][3:]
+            end_day = items[1][0:2]
+            end_hr = items[1][3:]
+
+            if "Common Time" in line:
+
+                # 2018 11 23 no longer scheduling twofsound as common time.
+                if "no switching" in line:
+                    mode_to_use = modes["no_switching_time"]
+                else:
+                    mode_to_use = modes["htr_common_time"]
+
+            if "Special Time" in line:
+
+                if "THEMIS" in line:
+                        mode_to_use = modes["themis_time"]
+
+                if "ST-APOG" in line:
+                    if radar.upper() in swg_lines[idx+1]:
+                        mode_to_use = modes["rbsp_time"]
+                    else:
+                        mode_to_use = modes["special_time_normal"]
+
+                if "ARASE" in line:
+                    if radar.upper() in swg_lines[idx+1]:
+                        if "themis" in swg_lines[idx+1]:
+                            mode_to_use = modes["themis_time"]
+                        if "interleaved" in swg_lines[idx+1]:
+                            mode_to_use = modes["interleaved_time"]
+                    else:
+                        mode_to_use = modes["special_time_normal"]
+
+
+                else:
+                    print("Unknown Special Time: using default common time")
+                    mode_to_use = modes["htr_common_time"]
+
+                # Skip next line
+                skip_line = True
+
+            if "Discretionary Time" in line:
+                mode_to_use = modes["discretionary_time"]
+
+            param = {"mmddyyyy": "{}/{}/{}".format(month, start_day, year),
+                     "hhmm" : "{}:00".format(start_hr),
+                     "experiment" : mode_to_use}
+            parsed_params.append(param)
+
+        return parsed_params
+
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Automatically schedules new events from the SWG")
+    parser.add_argument('--emails_filepath',required=True, help='A list of emails to send logs to')
+    parser.add_argument('--scd_dir', required=True, help='The scd working directory')
+
+    args = parser.parse_args()
+
+    scd_dir = args.scd_dir
+    scd_logs = scd_dir + "/logs"
+
+    emailer = email_utils.Emailer(args.emails)
```

Comment:
I think this needs to be updated for emails_filepath, you may have done already

Reply:
done.